awx/conf/views.py

```diff
@@ -161,40 +161,8 @@ class SettingLoggingTest(GenericAPIView):
     filter_backends = []
 
     def post(self, request, *args, **kwargs):
-        defaults = dict()
-        for key in settings_registry.get_registered_settings(category_slug='logging'):
-            try:
-                defaults[key] = settings_registry.get_setting_field(key).get_default()
-            except serializers.SkipField:
-                defaults[key] = None
-        obj = type('Settings', (object,), defaults)()
-        serializer = self.get_serializer(obj, data=request.data)
-        serializer.is_valid(raise_exception=True)
-        # Special validation specific to logging test.
-        errors = {}
-        for key in ['LOG_AGGREGATOR_TYPE', 'LOG_AGGREGATOR_HOST']:
-            if not request.data.get(key, ''):
-                errors[key] = 'This field is required.'
-        if errors:
-            raise ValidationError(errors)
-
-        if request.data.get('LOG_AGGREGATOR_PASSWORD', '').startswith('$encrypted$'):
-            serializer.validated_data['LOG_AGGREGATOR_PASSWORD'] = getattr(
-                settings, 'LOG_AGGREGATOR_PASSWORD', ''
-            )
-
-        try:
-            class MockSettings:
-                pass
-            mock_settings = MockSettings()
-            for k, v in serializer.validated_data.items():
-                setattr(mock_settings, k, v)
-            AWXProxyHandler().perform_test(custom_settings=mock_settings)
-            if mock_settings.LOG_AGGREGATOR_PROTOCOL.upper() == 'UDP':
-                return Response(status=status.HTTP_201_CREATED)
-        except LoggingConnectivityException as e:
-            return Response({'error': str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
-        return Response(status=status.HTTP_200_OK)
+        logging.getLogger('awx').info('AWX Connection Test')
```

Comment:
This doesn't give synchronous feedback, and requires you to save your logging settings before testing.

🤷‍♂

Reply:
This seems like a hard sell to me. Yes, I completely see the reason that we can't have a full systems test (unless rsyslog offers some compatible test command).

But confirming connectivity to the external service is still valuable to the user. Maybe we could "fake" it, by creating a new pure-python handler and trying to send it that way.

Reply:
> But confirming connectivity to the external service is still valuable to the user. Maybe we could "fake" it, by creating a new pure-python handler and trying to send it that way.

Maybe?

You're not testing the *actual* thing, though, so I'm not sure if I agree with that.

Reply:
For what it's worth, this is what we do with notification templates - you can't test them until after you've saved.